lambdaclass/ethrex
# feat(core): add metrics/observability with prometheus

**Motivation**

We need a starting point to create dashboards with Grafana. Metrics are essential to observe the behavior of the node.

**Description**

- Simple dashboard with a counter. Counts the transactions.
  - We neeed an API with the `/metrics` endpoint &rarr; using axum for fast development, we may want to create a simple API from scratch without any crate.

- To test it out:
1. `cd crates/l2 && cp .env.example .env`
2. `make rm-db-l2 && make init`

- In a new terminal:
1. `cd crates/l2 && make cli`
2. `ethrex_l2 test load --path <full_path>/ethrex/test_data/private_keys.txt -i 1`

- In the browser go to [http://localhost:3802/](http://localhost:3802/):
  - Login with `admin`/`admin`
  - Go to Dashboards &rarr; open the `Ethrex - Transactions` dashboard, the dashboard should show `172`. The previous cli's command sent 172 transactions.
  
Example output:

![image](https://github.com/user-attachments/assets/f585d850-490d-4133-8216-4f3d7e8c42ff)


Closes #1429
